src/lib/bugsnag/index.ts

```diff
@@ -0,0 +1,40 @@
+import Bugsnag from '@bugsnag/js';
+import BugsnagPluginReact from '@bugsnag/plugin-react';
+
+export function start() {
+	// next.js executes top-level code at build time. See https://github.com/vercel/next.js/discussions/16840 for further example
+	// So use NEXT_PHASE to avoid Bugsnag.start being executed during the build phase
+	// See https://nextjs.org/docs/api-reference/next.config.js/introduction and https://github.com/vercel/next.js/blob/canary/packages/next/shared/lib/constants.ts#L1-L5 for
+	// more details on NEXT_PHASE
+	if (process.env.NEXT_PHASE !== 'phase-production-build') {
```

Comment:
Should we have this listed somewhere with all possible values? Enum/Const/something...

Reply:
This is a Next.js constant, we can use the reference to it instead of having a magic string

```tsx
import { PHASE_PRODUCTION_BUILD } from 'next/dist/shared/lib/constants';
```

Reply:
@kristian240 is it ok to use `next/dist`?

Reply:
We might have a breaking change, with the import, but we also might have a breaking change with the hardcoded string. So I think it is okay to have it as an import.

Reply:
Found this, it seems like ti's exposed but not in TS definitions https://github.com/vercel/next.js/blob/canary/packages/next-plugin-storybook/preset.js#L1